### PR TITLE
[PREVIEW] Feature/sl 1881/ammending mulitsession request

### PR DESCRIPTION
**Pull request by andrewtaylorkainos**
### Change description ###
This PR required updating all actions so that ONLY the hearing parts for the hearing which 'can be withdrawn / unlisted / listed' are retrieved.
The update action has been amended so that when the number of sessions increases hearing parts are added; also when the number of sessions is decreased hearing parts are vacated/withdrawn depending on whether they were previously listed/unlisted.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/sandl/snlevents/actions/listingrequest/UpdateListingRequestAction.java [hunk 1]
```diff
@@ -90,6 +86,30 @@ public void getAndValidateEntities() {
...
+        if (!hearing.isMultiSession() && updateListingRequest.getNumberOfSessions() > 1) {
+            throw new SnlEventsException("Single-session hearings cannot have more than 2 sessions!");
+        }
+
+        if (hearing.isMultiSession() && updateListingRequest.getNumberOfSessions() < 2) {
+            throw new SnlEventsException("Multi-session hearings cannot have less than 2 sessions!");
+        }
+
+        if (hearing.getStatus().getStatus().equals(Status.Listed)) {
+            if (!OffsetDateTime.now().toLocalDate().isBefore(
```
[the remark this answers is not in the document]
Comment (andrewtaylorkainos) [reply src/main/java/uk/gov/hmcts/reform/sandl/snlevents/actions/listingrequest/UpdateListingRequestAction.java]: Probably not, was just ensuring that the two dates are converted to the same time Zone.
I suppose the time zone is already present in the OffsetDateTime so it'll make the comparison without converting it - removing the conversion :)

File: src/test/java/uk/gov/hmcts/reform/sandl/snlevents/actions/listingrequest/CreateListingRequestActionTest.java [hunk 2]
```diff
@@ -65,8 +65,8 @@
     @Spy
     private HearingMapper hearingMapper;
 
-    @Spy
-    private FactsMapper factsMapper;
+    @Rule
+    public ExpectedException expectedException = ExpectedException.none();
```
Comment (r-kruk-kainos) [inline src/test/java/uk/gov/hmcts/reform/sandl/snlevents/actions/listingrequest/CreateListingRequestActionTest.java]: Couldn't you just annotate tests like that     @Test(expected = SnlEventsException.class)
?

Comment (andrewtaylorkainos) [reply src/test/java/uk/gov/hmcts/reform/sandl/snlevents/actions/listingrequest/CreateListingRequestActionTest.java]: This rule actually give you the ability to extract the message thrown so we can check to see the correct error has been triggered

File: src/main/java/uk/gov/hmcts/reform/sandl/snlevents/actions/listingrequest/UpdateListingRequestAction.java [hunk 3]
```diff
@@ -102,22 +121,17 @@ public void getAndValidateEntities() {
...
-
-        hearing.getHearingParts().forEach(hp -> userTransactionDataList.add(new UserTransactionData("hearingPart",
-            hp.getId(),
-            null,
-            "lock",
-            "unlock",
-            0)));
-
-        return userTransactionDataList;
+        hearing.getHearingParts().forEach(hp ->
```
[the remark this answers is not in the document]
Comment (andrewtaylorkainos) [reply src/main/java/uk/gov/hmcts/reform/sandl/snlevents/actions/listingrequest/UpdateListingRequestAction.java]: Yep! done

File: src/main/java/uk/gov/hmcts/reform/sandl/snlevents/actions/listingrequest/UpdateListingRequestAction.java [hunk 4]
```diff
@@ -127,6 +141,94 @@ public UUID getUserTransactionId() {
...
+            .map(HearingPart::getSession)
+            .filter(Objects::nonNull)
+            .collect(Collectors.toList());
+    }
+
+    private void setUpdatedHearingValues() {
+        hearing.setCaseNumber(updateListingRequest.getCaseNumber());
+        hearing.setCaseTitle(updateListingRequest.getCaseTitle());
+        CaseType caseType = caseTypeRepository.findOne(updateListingRequest.getCaseTypeCode());
+        hearing.setCaseType(caseType);
```
[the remark this answers is not in the document]
Comment (andrewtaylorkainos) [reply src/main/java/uk/gov/hmcts/reform/sandl/snlevents/actions/listingrequest/UpdateListingRequestAction.java]: Done

**Pull request closed without merging**